api/src/backend_common/auth.py

```diff
@@ -569,12 +569,15 @@ def get_taskcluster_credentials():
         return dict(
             client_id='XXX',
             access_token='YYY',
+            root_url='http://ZZZ'
         )
     return dict(
         client_id=os.environ.get('TASKCLUSTER_CLIENT_ID',
                                  flask.current_app.config.get('TASKCLUSTER_CLIENT_ID')),
         access_token=os.environ.get('TASKCLUSTER_ACCESS_TOKEN',
                                     flask.current_app.config.get('TASKCLUSTER_ACCESS_TOKEN')),
+        root_url=os.environ.get('TASKCLUSTER_ROOT_URL',
+                                flask.current_app.config.get('TASKCLUSTER_ROOT_URL')),
```

Comment:
I'm curious how
- defaulting to the environment interacts with potentially setting an environment variable to control where to get the secrets from (the production cluster is migrating on Nov 9, so this will become more relevant soon).
- relatedly, will having a client_id/access_token for a different cluster have an impact (I presume the credentials will be fore the cluster the secret is fetched from, not the cluster the auth is against.

Reply:
yeah, that is true. i guess the easier would be to just ignore the os.environment in auth.py and rely on `flask.current_app.config`. then we set client_id/access_token/root_url to fetch secrets and there we can reconfigure client_id/access_token/root_url